src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -135,14 +171,20 @@ object GenomeReference {
   def gen: Gen[GenomeReference] = for {
     name <- Gen.identifier
     nContigs <- Gen.choose(3, 50)
-    contigs <- Gen.distinctBuildableOfN[Array, Contig](nContigs, Contig.gen)
-    contigNames = contigs.map(_.name).toSet
-    xContig <- Gen.oneOfSeq(contigNames.toSeq)
-    yContig <- Gen.oneOfSeq((contigNames - xContig).toSeq)
-    mtContig <- Gen.oneOfSeq((contigNames - xContig - yContig).toSeq)
+    contigs <- Gen.distinctBuildableOfN[Array, String](nContigs, Gen.identifier)
+    lengths <- Gen.distinctBuildableOfN[Array, Int](nContigs, Gen.choose(1000000, 500000000))
+    xContig <- Gen.oneOfSeq(contigs)
+    yContig <- Gen.oneOfSeq((contigs.toSet - xContig).toSeq)
+    mtContig <- Gen.oneOfSeq((contigs.toSet - xContig - yContig).toSeq)
```

Comment:
Ah, I see now, those tests are already in master. You should use plural there on contigs or change the form to something like:
`The following contig names were found in both X and Y contigs:`

And I think `require` should be `fatal`.